lib/_http_outgoing.js

```diff
@@ -733,6 +737,9 @@ function write_(msg, chunk, encoding, callback, fromEnd) {
     throw new ERR_STREAM_NULL_VALUES();
   } else if (typeof chunk === 'string') {
     len = Buffer.byteLength(chunk, encoding);
+    if (msg._contentLength && len > msg._contentLength) {
+      throw new ERR_CONTENT_LENGTH_HEADER();
+    }
```

Comment:
You need to check this for strings as well as buffers and also you need to check against **the total number of bytes written so far**.

Reply:
Thank, I would make more tests and do the changes